flow/core/kernel/energy.py

```diff
@@ -0,0 +1,119 @@
+import dill as pickle
+import boto3
+import botocore
+import math
+import random
+import statistics
+import numpy as np
+from scipy.interpolate import interp1d
+from collections import namedtuple
+
+
+class EnergyModel():
+    """Base energy model class."""
+
+    def __init__(self,kernel):
+        self.vehicle = vehicle
+
+    def get_energy(self, veh_id):
+        #return self.calc_energy(veh_id)
+        pass
+
+
+class PowerDemandModel(EnergyModel):
```

Comment:
Not necessary, but I think we could move this and other child classes to other file(s). Maybe vehicle power demand in its own file, and then all Toyota models in one file?

also remember to implement an "EV" version and a "Gasoline" version of the vehicle power demand model